File: net/importing-to-kentico-cloud/UpsertLanguageVariant.cs
// DocSection: importing_upsert_variant
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.UpsertLanguageVariantAsync(identifier, new LanguageVariantUpsertModel
{
    Elements = new dynamic[]
    {
        new TextElement
        {
            Element = Reference.ByExternalId("street"),
            Value = "Nove Sady 25",
        }.ToDynamic(),
        new TextElement
        {
            Element = Reference.ByExternalId("city"),
            Value = "Brno",
        }.ToDynamic(),
        new TextElement
        {
            Element = Reference.ByExternalId("country"),
            Value = "Czech republic",
        }.ToDynamic(),
        new TextElement
        {
            Element = Reference.ByExternalId("state"),
            Value = "Jihomoravsky kraj",
        }.ToDynamic(),
        new TextElement
        {
            Element = Reference.ByExternalId("zip_code"),
            Value = "60200",
        }.ToDynamic(),
        new TextElement
        {
            Element = Reference.ByExternalId("phone"),
            Value = "+420 555 555 555",
        }.ToDynamic(),
        new TextElement
        {
            Element = Reference.ByExternalId("email"),
            Value = "brnocafe@kontent.ai",
        }.ToDynamic(),
    }
});
// EndDocSection

File: js/importing-to-kentico-kontent/UpsertLanguageVariant.js
// DocSection: importing_upsert_variant
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

client.upsertLanguageVariant()
  .byItemExternalId('ext-cafe-brno')
  .byLanguageCodename('en-US')
  .withElements([
    {
      element: {
        codename: 'street'
      },
      value: 'Nove Sady 25'
    },
    {
      element: {
        codename: 'city'
      },
      value: 'Brno'
    },
    {
      element: {
        codename: 'country'
      },
      value: 'Czech Republic'
    },
    {
      element: {
        codename: 'state'
      },
      value: 'Jihomoravsky kraj'
    },
    {
      element: {
        codename: 'zip_code'
      },
      value: '60200'
    },
    {
      element: {
        codename: 'phone'
      },
      value: '+420 555 555 555'
    },
    {
      element: {
        codename: 'email'
      },
      value: 'brnocafe@kontent.ai'
    }
  ])
  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
    (error) => {
      console.log(error);
    });
// EndDocSection